### PR TITLE
FileManager.fileExists(atPath:) should follow symlinks on Windows

### PR DESCRIPTION
**Pull request by jmschonfeld**
As a followup to https://github.com/apple/swift-foundation/pull/858, I've also noticed that `FileManager.fileExists(atPath:)` does not correctly handle symlinks on Windows. On Darwin/Linux this function determines if the destination of the symlink exists, but on Windows it currently checks whether the symlink itself exists. This updates the logic to follow the symlink and check the destination instead to match previous (swift 5.10) Windows behavior (and behavior on Darwin/Linux)

### CODE REVIEW

**Comment by jmschonfeld** [comment]
@swift-ci please test

**Comment by jmschonfeld** [comment]
This PR depends on changes from https://github.com/apple/swift-foundation/pull/858, I'll wait until that is merged and then rebase and re-run tests

**Review by itingliu — APPROVED** [review]

**Comment by jmschonfeld** [comment]
@swift-ci please test

**Pull request merged**
